packages/react/src/component.tsx

```diff
@@ -29,11 +31,23 @@ export function Uploader<TRouter extends FileRouter>(
   );
 }
 
-export function generateComponents<TRouter extends FileRouter>() {
+export function generateComponents<TRouter extends FileRouter>(initOpts?: {
+  /**
+   * The URL where you expose your UploadThing router.
+   * @default `/api/uploadthing`
+   */
+  url: string;
+}) {
```

Comment:
having this optional makes for quite a few codepaths having weird optional chains / nullish coalescing. to me it makes sense that this is required, but it would be a breaking change that would require a major bump

Reply:
I am on with doing a major for this. I agree that it makes sense as required.